### ISSUE TITLE
compgen -V array not considered assigned

### ISSUE THREAD

**Issue by rieje** [koalaman/shellcheck#3466]
#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows a potential problem:

```sh
#!/bin/bash
compgen -V files -G '*'
printf "%s\n" "${files[@]}"
```

#### Here's what shellcheck currently says:
SC1254: files is referenced but not assigned

#### Here's what I wanted to see:
No warning, `files` was assigned.

**Comment by matt-kita** [comment]
> #!/bin/bash
> compgen -V array not considered assigned
> [...]
> No warning, `array` was assigned.

Aren't you actually misusing the `-V` flag in the first place?
According to https://www.gnu.org/software/bash/manual/bash.html (for Bash version 5.3, 18 May 2025, at the time of writing this)

```
compgen

    compgen [-V varname] [option] [word]
[...]
If the -V option is supplied, compgen stores the generated completions into the indexed array 
variable varname instead of writing them to the standard output.
```

**Comment by rieje** [comment]
Whoops the example is wrong, I updated it. Basically, `compgen -V` is a Bash builtin way to to store glob-expanded files into the array.

**Comment by matt-kita** [comment]
> compgen -V files -G '*'

Sorry to be nitpicking, but to store a list of files located in your current working directory, you don't need to involve the entire bash completion facility and `compgen` builtin, running just `shopt -s nullglob; files=(*)` would most likely give the expected results ;)

Nevertheless, having a `completions` array indicated as "not assigned" after `compgen -V completions` is indeed a valid point.
